src/datasource.js

```diff
@@ -176,7 +176,13 @@ export class HawkularDatasource {
       return flatTags.map(tag => {
         return {text: tag, value: tag};
       });
-    });
+    }));
+  }
+
+  runWithResolvedVariables(target, func) {
+    const resolved = this.variables.resolve(target, {});
+    return this.q.all(resolved.map(p => func(p)))
+      .then(result => [].concat.apply([], result));
```

Comment:
This writing "`[].concat.apply([], result)`" is what I've found for googling "javascript flatMap"
Basically, `result` is something like `[[a,b,c], [d,e,f]]` and I want to flat it down to `[a,b,c,d,e,f]`, ie. "flatMap" in some other languages.
Do you know a better way to do it? I don't think the spread operator is designed for that, is it?